app/test_main.py

```diff
@@ -1 +1,27 @@
-# write your code here
+from typing import Callable
+from unittest import mock
+
+import pytest
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, expected_url, connection, expected_result",
+    [
+        pytest.param("https://www.google.com/", True, False, "Not accessible"),
+        pytest.param("https://www.google.com/", False, True, "Not accessible"),
+        pytest.param("https://www.google.com/", True, True, "Accessible")
```

Comment:
could u move ur urls directly to the func ? cause u are using the same url every time 
also what about False/False case ?